packages/svelte/src/reactivity/set.js

```diff
@@ -65,12 +50,17 @@ export class ReactiveSet extends Set {
 		var proto = ReactiveSet.prototype;
 		var set_proto = Set.prototype;
 
-		for (var method of read) {
+		for (var method of methods) {
 			// @ts-ignore
 			proto[method] = function (...v) {
 				get(this.#version);
```

Comment:
we don't need this for `difference`, `union` etc. i think we should put `forEach` on the class instead of in the `methods` list, remove `get(...)` here, remove `if (val instanceof Set)` below, and wrap the whole thing in `if (method in set_proto)` since it's not universally supported

Reply:
These methods can still be used in a reactive context. Should we leave this as is then?

> remove if (val instanceof Set) below, and wrap the whole thing in if (method in set_proto) since it's not universally supported

Not sure what you mean by this. How will we know the return value is a Set? What is not universally supported?

Reply:
yeah, we can still check the version, i guess someone could do `{set.union(other).size}` or something.

the methods in question are new and not everyone implements them:

<img width="531" alt="image" src="https://github.com/sveltejs/svelte/assets/1162160/8d5ad372-b7ff-4a57-be98-48bbcb5b233e">

Reply:
> How will we know the return value is a Set?

What I'm saying is that we can safely assume that the argument to all the methods other than `forEach` is set-like (which isn't actually the same as `instanceof Set` — see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set#set-like_objects), so it makes sense to just put `forEach` on the class since it's the outlier

Reply:
I updated the PR. However I kept `forEach` with the other non set like methods (they instead return booleans)